### ISSUE TITLE
SOR cannot find pool ID

### ISSUE THREAD

**Issue by xianny** [balancer-labs/balancer-sor#29]
When trying to use `sor.smartOrderRouter`, it mis-computes the pool ID and fails.

Error log:

```
TypeError: Cannot read property 'spotPrice' of undefined
    at /Users/xianny/src/balancer/node_modules/@balancer-labs/sor/dist/sor.js:250:58
    at Array.forEach (<anonymous>)
    at getInputAmountsForEp (/Users/xianny/src/balancer/node_modules/@balancer-labs/sor/dist/sor.js:246:10)
    at /Users/xianny/src/balancer/node_modules/@balancer-labs/sor/dist/sor.js:22:21
    at Array.forEach (<anonymous>)
    at Object.exports.smartOrderRouter (/Users/xianny/src/balancer/node_modules/@balancer-labs/sor/dist/sor.js:19:19)
    at sor.getPoolsWithTokens.then.data (/Users/xianny/src/balancer/test.js:14:26)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```

Code lifted from https://docs.balancer.finance/sor/development#example

```
// test.js
// dependencies": {
//     "@balancer-labs/sor": "^0.2.7",
//     "bignumber.js": "^9.0.0",
//     "ethers": "^4.0.47"
// }

const sor = require('@balancer-labs/sor');
const BigNumber = require('bignumber.js');
const ethers = require('ethers');


const MAX_UINT = ethers.constants.MaxUint256;

// MAINNET
const tokenIn = '0x6B175474E89094C44Da98b954EedeAC495271d0F' // DAI
const tokenOut = '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2' // WETH

sor.getPoolsWithTokens(tokenIn, tokenOut).then(data => {
    const poolData = sor.parsePoolData(data.pools, tokenIn, tokenOut);
    const sorSwaps = sor.smartOrderRouter(
        poolData,
        'swapExactIn',
        new BigNumber('10000000000000000000'),
        new BigNumber('10'),
        0
    );

});
```

Error is being thrown when [this line](https://github.com/balancer-labs/balancer-sor/blob/master/src/sor.ts#L387-L389) returns `balancer = undefined`:

**Comment by mikemcdonald** [comment]
Thanks for the details @xianny. Can you elaborate more on what you mean by it mis-computes the pool ID, or what poolData can generate this error? I'm able to run the above code snippet successfully using the dependency versions listed. (Improving some of the error handling and checks is on the to-do list)

Btw, we usually recommend teams use `parsePoolDataOnchain` to remove any dependencies on subgraph balances.

**Comment by FernandoMartinelli** [comment]
I'm investigating this bug. It seems to be a very rare edge case caused by
a rounding error. I'll keep you posted.

**Comment by mikemcdonald** [comment]
@xianny this has been fixed in @balancer-labs/sor 0.2.8. Thanks for reporting
